packages/preferences/src/browser/views/components/preference-select-input.ts

```diff
@@ -17,29 +17,60 @@
 import { PreferenceLeafNodeRenderer, PreferenceNodeRenderer } from './preference-node-renderer';
 import { injectable, interfaces } from '@theia/core/shared/inversify';
 import { JSONValue } from '@theia/core/shared/@phosphor/coreutils';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { SelectComponent, SelectOption } from '@theia/core/lib/browser/widgets/select-component';
 import { Preference } from '../../util/preference-types';
 import { PreferenceLeafNodeRendererContribution } from './preference-node-renderer-creator';
+import * as React from '@theia/core/shared/react';
+import * as ReactDOM from '@theia/core/shared/react-dom';
 
 @injectable()
-export class PreferenceSelectInputRenderer extends PreferenceLeafNodeRenderer<JSONValue, HTMLSelectElement> {
+export class PreferenceSelectInputRenderer extends PreferenceLeafNodeRenderer<JSONValue, HTMLDivElement> {
+
+    protected readonly onDidChangeEmitter = new Emitter<number>();
+
+    protected get onDidSelectedChange(): Event<number> {
+        return this.onDidChangeEmitter.event;
+    }
```

Comment:
Are these used?

Reply:
They are used to notify the select component if the preference has been changed from the outside (i.e. by modifying the JSON). See:

https://github.com/eclipse-theia/theia/blob/26e29077154b8864a4aabf0b0a8c746d6fca2464/packages/preferences/src/browser/views/components/preference-select-input.ts#L66-L71

Reply:
I had mistakenly reversed the flow: I thought the `SelectComponent`'s `onDidSelectionChange` was to notify external parties, not to receive notifications. In principle, I'd probably prefer a more DOM-like API (i.e. the ability to set the value), but with React, that would mean tracking a ref anyway, so I'm not sure it would end up any cleaner. What do you think?

Reply:
Since this is the only case where we need to programmatically change the selection value (so far), I would actually go with the ref idea.